### ISSUE TITLE
Numeric sequence order follows line selection order

### ISSUE THREAD

**Issue by JessePeden** [volkerdobler/insertnums#11]
With only the default settings immediately after installing the extension, trying to insert a numeric sequence beginning at 1 and stepping 1, the numeric sequencing begins at whichever line was first selected and ends with the last line selected, potentially resulting in improper sequencing.  If you select your lines in the exact order that you want them numbered, you won't have any issues, but if you click (for example) line 1, 3, 4, 6, 5, 2 and then insert a sequence at the end of those lines, beginning with the integer 1 and stepping 1, you end up with something that looks like the following:

```
<line 1>1
<line 2>6
<line 3>2
<line 4>3
<line 5>5
<line 6>4
```

The way that it _should_ behave is that the numeric sequencing starts at the the first line in the list, which may not be the line that was first clicked/selected (like if you happen to be halfway down a document and then decide to change all occurrences of a word to a sequence, etc. and your last clicked line is halfway down the page where you've been working at most recently).

**Comment by volkerdobler** [comment]
Hi sybercorp,
thanks for your message. I think, what you need is already there (even, that the documentation is not detailed to this point).
Down in the "Syntax details" section, you can read:
`$ the selections will be "sorted" (without this option, new chars will be inserted in the order of the multiline clicks)`
which means, if you put a $ at the end of your expression, the order of the inserted numbers will not follow the multi-selection, but from the first to the last selection. Which is - what I assume - what you need.
Unfortunately, there is not "default" switch at the moment to implement this behaviour as standard. Will think about it for the next version - thanks for the suggestion and reminder of a missing point in the documentation.
Regards,
Volker

**Comment by JessePeden** [comment]
@volkerdobler Thank you.  Yes, that's exactly the issue I'm facing.  I'll try to remember that there's an operator to "correct" the behavior to how I need it to work but, yes, a way to set that option at a default level would be so much better than having to remember the operator (or whether it goes at the start or end of the command).

**Comment by volkerdobler** [comment]
> @volkerdobler Thank you. Yes, that's exactly the issue I'm facing. I'll try to remember that there's an operator to "correct" the behavior to how I need it to work but, yes, a way to set that option at a default level would be so much better than having to remember the operator (or whether it goes at the start or end of the command).

Hi @JessePeden, I just released version 0.9.4. It finally includes the option to change the default behavior with a config switch. Look at "insertOrder" and switch it from default (cursor) to "sorted". This will insert all values in a sorted order and not based on your click order.
Hope this helps and enjoy the extension

**Comment by JessePeden** [comment]
> > @volkerdobler Thank you. Yes, that's exactly the issue I'm facing. I'll try to remember that there's an operator to "correct" the behavior to how I need it to work but, yes, a way to set that option at a default level would be so much better than having to remember the operator (or whether it goes at the start or end of the command).
> 
> 
> 
> Hi @JessePeden, I just released version 0.9.4. It finally includes the option to change the default behavior with a config switch. Look at "insertOrder" and switch it from default (cursor) to "sorted". This will insert all values in a sorted order and not based on your click order.
> 
> Hope this helps and enjoy the extension

Thank you. 😀

**Comment by JessePeden** [comment]
Just now getting around to looking at this new ability, and I forked the project to make some spelling corrections to the descriptions in packages.json.  There's a Pull Request waiting for your approval.
